sebs/fission/fission.py

```diff
@@ -0,0 +1,373 @@
+            except subprocess.CalledProcessError:
+                logging.info(f"Creating env {name} failed. Retrying...")
+                sleep(10)
+                try:
+                    subprocess.run(
+                        f"fission env create --name {name} --image {image} \
+                        --builder {builder}".split(),
+                        check=True,
+                        stdout=subprocess.DEVNULL,
+                    )
+                except subprocess.CalledProcessError:
+                    self.storage.storage_container.kill()
+                    logging.info("Minio stopped")
+                    self.initialize()
+                    self.create_env_if_needed(name, image, builder)
+
+    def create_function(self, name: str, env_name: str, path: str):
+        packageName = f"{name}-package"
+        self.createPackage(packageName, path, env_name)
+        self.createFunction(packageName, name)
+
+    def createPackage(self, packageName: str, path: str, envName: str) -> None:
+        logging.info(f"Deploying fission package...")
+        self.packageName = packageName
+        try:
+            packages = subprocess.run(
+                "fission package list".split(), stdout=subprocess.PIPE, check=True
+            )
+            subprocess.run(
+                f"grep {packageName}".split(),
+                check=True,
+                input=packages.stdout,
+                stdout=subprocess.DEVNULL,
+            )
+            logging.info("Package already exist")
+        except subprocess.CalledProcessError:
+            process = f"fission package create --sourcearchive {path} \
+            --name {packageName} --env {envName} --buildcmd ./build.sh"
+            subprocess.run(process.split(), check=True)
+            logging.info("Waiting for package build...")
+            while True:
+                try:
+                    packageStatus = subprocess.run(
+                        f"fission package info --name {packageName}".split(),
+                        stdout=subprocess.PIPE,
+                    )
+                    subprocess.run(
+                        f"grep succeeded".split(),
+                        check=True,
+                        input=packageStatus.stdout,
+                        stderr=subprocess.DEVNULL,
+                    )
+                    break
+                except subprocess.CalledProcessError:
+                    if "failed" in packageStatus.stdout.decode("utf-8"):
+                        logging.error("Build package failed")
+                        raise Exception("Build package failed")
+                    sleep(3)
+                    continue
+            logging.info("Package ready")
+
+    def deletePackage(self, packageName: str) -> None:
+        logging.info(f"Deleting fission package...")
+        subprocess.run(f"fission package delete --name {packageName}".split())
+
+    def createFunction(self, packageName: str, name: str) -> None:
```

Comment:
The class should implement the abstract functions from `sebs.faas.system.System`

Reply:
yes, I have changed that